core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -362,25 +365,46 @@ trait WorkspaceComponent {
       query.result.map(_.map { case (workspace, accessLevel) => WorkspacePermissionsPair(workspace.id.toString(), WorkspaceAccessLevels.withName(accessLevel.accessLevel)) })
     }
 
-    def findWorkspacesForGroup(group: RawlsGroupRef): ReadAction[Seq[Workspace]] = {
-      val byAccessGroupAction: ReadAction[Seq[WorkspaceRecord]] = for {
-        groupRecs <- rawlsGroupQuery.findGroupByName(group.groupName.value).result
-        allGroups <- rawlsGroupQuery.listParentGroupsRecursive(groupRecs.toSet, groupRecs.toSet)
-        workspaceRecs <- findWorkspacesForGroups(allGroups).result
-      } yield workspaceRecs
+    /**
+     * Find the intersection groups that need to be recomputed given the change in group and the groups that
+     * should be used in the intersection
+     * @param group group that has changed to trigger the recompute
+     * @return
+     */
+    def findAssociatedGroupsToIntersect(group: RawlsGroupRef): ReadAction[Seq[(GroupsToIntersect)]] = {
```

Comment:
this implementation is optimized from findWorkspacesForGroup. 

The old way would load each workspace (which in itself is a bunch of queries) then for each of those call updateIntersectionGroupMembers which loaded each realm group individually (3 or so queries per).

The new way does not load the workspace or groups. All we need are the group references now. We need to know which group refs to update and which to intersect.